python/planout/experiment.py

```diff
@@ -69,7 +70,11 @@ def __init__(self, **inputs):
     def _assign(self):
         """Assignment and setup that only happens when we need to log data"""
         self.configure_logger()  # sets up loggers
-        self.assign(self._assignment, **self.inputs)
+
+        #consumers can optionally return False from assign if they don't want exposure to be logged
+        assign_val = self.assign(self._assignment, **self.inputs)
+        if self._in_experiment is self._default_in_experiment_val:
```

Comment:
this still allows consumers to manually set self._in_experiment in the assign function as is done in the tests and only overrides it if _in_experiment is not set to the default value.